src/Openbuildings/Spiderling/Driver/Phantomjs/Connection.php

```diff
@@ -269,4 +287,66 @@ protected function call($command, array $options = array())
 
 		return $result;
 	}
+
+	/**
+	 * Start a phantomjs server in the background. Set port, server js file, additional files and log file.
+	 *
+	 * @param  string  $file       the server js file
+	 * @param  integer $port       the port to start the server on
+	 * @param  string  $additional additional file, passed to the js server
+	 * @param  string  $log_file
+	 * @return string the pid of the newly started process
+	 * @throws Exception
+	 */
+	protected function _start($file, $port, $additional = NULL, $log_file = '/dev/null')
+	{
+		if ( ! Network::is_port_open('localhost', $port))
+			throw new Exception('Port :port is already taken', array(':port' => $port));
+
+		if ($log_file !== '/dev/null' AND ! is_file($log_file))
+			throw new Exception('Log file (:log_file) must be a file or /dev/null', array(':log_file' => $log_file));
+
+		return shell_exec(strtr('nohup :command > :log 2> :log & echo $!', array(
+			':command' => $this->_command($file, $port, $additional),
+			':log' => $log_file,
+		)));
+	}
+
+	/**
+	 * kill a server on a given pid
+	 * @param  string $pid
+	 */
+	protected function _kill($pid)
+	{
+		shell_exec('kill '.$pid);
+	}
+
+	/**
+	 * Return the command to start the phantomjs server
+	 *
+	 * @param  string  $file       the server js file
+	 * @param  integer $port
+	 * @param  string  $additional additional js file
+	 * @return string
+	 * @throws Exception
+	 */
+	protected function _command($file, $port, $additional = NULL)
+	{
+		$dir = realpath(__DIR__.'/../../../assets').'/';
```

Comment:
I think this should use `__DIR__.`/../../../../../assets'` to find the assets folder. Haven't tested it, but I can see the tests fail, because the file in `assets` is not found. The `realpath()` method returns `FALSE` when the folder doesn't exist and the file we search for then is `/phantom.js` as stated here: https://travis-ci.org/OpenBuildings/spiderling/jobs/164030465#L262

Also the whole thing could be combined in `$file = realpath(__DIR__."/../../../../../assets/$file")`. `realpath()` could resolve files as well so no need to do several concatenations.